CONTRIBUTING.md

```diff
@@ -270,6 +270,31 @@ code, run all of the slow tests, and also calculate code coverage, you could run
 pytest --cov=pyquil --use-seed=False --runslow <path/to/test-file-or-dir>
 ```
 
+#### Benchmarks
+
+We use benchmarks to ensure the performance of pyQuil is tracked over time, preventing unintended 
+regressions. Benchmarks are written and run using [pytest-benchmark](https://pytest-benchmark.readthedocs.io/en/latest/).
+This plugin provides a fixture called `benchmark` that can be used to benchmark a Python function.
+
+For organization, all benchmarks are located in the `test/benchmarks` directory. To run the 
+benchmarks, use the command:
+
+```
+pytest -v test/benchmarks  # or use the Makefile: `make bench`
+```
+
+Note that benchmark results are unique to your machine. They can't be directly compared to benchmark
+results on another machine unless it's a machine with identical specifications running in a similar
+environment. To track performance over time in a controlled way, we use _continuous benchmarking_.
+When a PR is opened, CI will run the benchmarks and compare the results to the most recent results
+on the `master` branch. Since CI always uses the same image and workflow, the results are fair
+to compare. When opening or reviewing a PR, you should evaluate the results and ensure there are no
```

Comment:
> Since CI always uses the same image and workflow, the results are fair
to compare.

Not necessarily true, as CI runners share resources, right? One run could have more resource contention than another, and appear to be slower.

Reply:
When I wrote that, I knew it would be subject to scrutiny 😄. It is fair though. I'll clarify that it is about as fair as can be reasonably expected.

Reply:
This thread was marked resolved, but I don't see any related changes.

Reply:
Oops, only committed it locally, it's pushed up now.